python/ray/serve/_private/router.py

```diff
@@ -358,6 +388,33 @@ def target_num_scheduling_tasks(self) -> int:
     def curr_replicas(self) -> Dict[str, ReplicaWrapper]:
         return self._replicas
 
+    def _get_actor_name_GCS_UNSAFE(self) -> str:
+        """Gets the name of the actor where this scheduler runs.
+
+        NOTE: this call hangs when the GCS is down. As long as this method is
+        called only when the scheduler is initialized, that should be
+        okay because a ServeHandle (and its scheduler) relies
+        on the Serve controller for intialization, and the Serve controller
+        is runs only when the GCS is up.
+
+        Return:
+            The name of the actor where this scheduler runs. If the scheduler
+            runs outside an actor or the call fails, returns an empty string.
+        """
+
+        if in_ray_driver_process_GCS_UNSAFE():
+            return ""
```

Comment:
let's return a custom string here for better observability. can use `"DRIVER"` maybe? that way the metrics are still reasonable